jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/AsyncUtil.java

```diff
@@ -56,4 +59,17 @@ public static <T> Function<Predicate<T>, CompletableFuture<T>> await(Supplier<T>
       return ret;
     });
   }
+
+  public static <T> T get(CompletableFuture<T> completableFuture, long timeout, TimeUnit timeUnit) {
```

Comment:
nit: maybe next time use `Duration` which is clearer

Reply:
Umm, how do I pass Duration into `get` method?

Reply:
you extract the relevant parts (It doesn't have to be a 1-1 mapping): 
e.g. `get($duration.toMillis(), TimeUnit.MILLISECONDS)`